homeassistant/components/light/ads.py

```diff
@@ -0,0 +1,119 @@
+SUPPORT_ADS = SUPPORT_BRIGHTNESS
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_ADSVAR): cv.string,
+    vol.Optional(CONF_ADSVAR_BRIGHTNESS): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the light platform for ADS."""
+    ads_hub = hass.data.get(DATA_ADS)
+    if not ads_hub:
+        return False
+
+    varname_enable = config.get(CONF_ADSVAR)
+    varname_brightness = config.get(CONF_ADSVAR_BRIGHTNESS)
+    devname = config.get(CONF_NAME)
+
+    add_devices([AdsLight(ads_hub, varname_enable, varname_brightness,
+                          devname)], True)
+
+
+class AdsLight(Light):
+    """Representation of ADS light."""
+
+    def __init__(self, ads_hub, varname_enable, varname_brightness, devname):
+        """Initialize AdsLight entity."""
+        self._ads_hub = ads_hub
+        self._on_state = False
+        self._brightness = 50
+        self._devname = devname
+        self.varname_enable = varname_enable
+        self.varname_brightness = varname_brightness
+        self.stype = 'dimmer'
+
+    @property
+    def name(self):
+        """Return the name of the device if any."""
+        return self._devname
+
+    @property
+    def brightness(self):
+        """
+        Brightness of the light.
+
+        This method is optional. Removing it indicates to Home Assistant
+        that brightness is not supported for this light.
+
+        """
+        return self._brightness
+
+    @property
+    def is_on(self):
+        """If light is on."""
+        return self._on_state
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_ADS
+
+    def turn_on(self, **kwargs):
+        """Turn the light on or set a specific dimmer value."""
+        brightness = kwargs.get(ATTR_BRIGHTNESS)
+        if brightness is not None:
+            self._brightness = brightness
+
+        bval = math.floor(self._brightness)
+
+        self._ads_hub.write_by_name(self.varname_enable, True,
+                                    self._ads_hub.PLCTYPE_BOOL)
+
+        if self.varname_brightness is not None:
+            self._ads_hub.write_by_name(self.varname_brightness, bval,
+                                        self._ads_hub.PLCTYPE_UINT)
+
+    def turn_off(self, **kwargs):
+        """Turn the light off."""
+        brightness = kwargs.get(ATTR_BRIGHTNESS)
+        if brightness is not None:
+            self._brightness = brightness
+        bval = math.floor(self._brightness)
+        self._ads_hub.write_by_name(self.varname_enable, False,
+                                    self._ads_hub.PLCTYPE_BOOL)
+        if self.varname_brightness is not None:
+            self._ads_hub.write_by_name(self.varname_brightness, bval,
+                                        self._ads_hub.PLCTYPE_UINT)
+
+    def value_changed(self, val):
```

Comment:
This doesn't seem to be used. Should this be registered as a callback?

Reply:
I thought this function would get called by homeassistant if the user changed a value in the frontend. But I just tried the platform without the function and the values get updated as well. So I removed it.

Reply:
Since you will use callbacks and not polls maybe you should turn it into a callback?

Reply:
Yes, done so.